distribution/docker/Dockerfile

```diff
@@ -0,0 +1,33 @@
+FROM maven:3-jdk-8 as builder
+
+COPY . /src
+WORKDIR /src
+RUN mvn install -ff -DskipTests -Dforbiddenapis.skip=true -Pdist -Pbundle-contrib-exts
```

Comment:
(minor)

You can use [cache mounts](https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/experimental.md) to make this go faster.

Something like 

```
RUN --mount=type=cache,id=druid-m2-repo,target=/root/.m2/repository mvn install -ff -DskipTests -Dforbiddenapis.skip=true -Pdist -Pbundle-contrib-exts -f /src/pom.xml
```

Reply:
I'd prefer not to do this. Its extremely new (released in Docker 18.09) so this will cause trouble in e.g. CI. Also it appears to not universally function with private registry:
https://github.com/moby/moby/issues/32507